api/component_data.go

```diff
@@ -0,0 +1,149 @@
+package api
+
+import (
+	"net/http"
+	"os"
+	"path/filepath"
+	"time"
+
+	"github.com/pkg/errors"
+)
+
+type ComponentDataService struct {
+	client *Client
+}
+
+type ComponentDataInitialRequest struct {
+	Name             string          `json:"name"`
+	Tags             []string        `json:"tags"`
+	SupportedMethods []string        `json:"supportedMethods"`
+	Documents        []*DocumentSpec `json:"documents"`
+}
+
+type DocumentSpec struct {
+	Name string `json:"name"`
+	Size int64  `json:"size"`
+}
+
+type InitialResponseRaw struct {
+	Data *InitialResponse `json:"data,omitempty"`
+}
+
+type InitialResponse struct {
+	Guid          string          `json:"guid,omitempty"`
+	UploadMethods []*UploadMethod `json:"uploadMethods,omitempty"`
+}
+
+type UploadMethod struct {
+	Method string            `json:"method,omitempty"`
+	Info   map[string]string `json:"info,omitempty"`
+}
+
+type CompleteRequest struct {
+	Guid string `json:"guid"`
+}
+
+type CompleteResponseRaw struct {
+	Data *CompleteResponse `json:"data,omitempty"`
+}
+
+type CompleteResponse struct {
+	Guid string `json:"guid,omitempty"`
+}
+
+func (svc *ComponentDataService) UploadFiles(name string, tags []string, paths []string) (string, error) {
+	initialRequest, err := buildInitialRequest(name, tags, paths)
```

Comment:
can you make this function names more explicit? `buildInitialRequest()` sounds very generic, an idea could be `componentDataBuildInitialRequest()` or something like that.

Reply:
Sure, done.